packages/hooks/src/useVerifyEmail.ts

```diff
@@ -1,14 +1,13 @@
-import { useWS } from '@deriv/api';
+import { useRequest } from '@deriv/api';
 import { useStore } from '@deriv/stores';
-import type { TSocketEndpoints } from '@deriv/api/types';
 import useCountdown from './useCountdown';
 
 const RESEND_COUNTDOWN = 60;
 
-export type TEmailVerificationType = TSocketEndpoints['verify_email']['request']['type'];
-
-const useVerifyEmail = (type: TEmailVerificationType) => {
-    const WS = useWS('verify_email');
+const useVerifyEmail = (
```

Comment:
If understood correctly, we're gonna use the same structure for other hooks as well, right? 
if so, can we have a generic type for our custom hooks parameters?
if it's not possible or it's too soon to add just ignore this, but please clarify this, thanks man :pray:

Reply:
@mohsen-deriv Actually I'm a bit unsure about it now, The idea was that if some endpoint needs some data we can abstract it into a hook so we don't have to write the same thing again and again. For example, the `verify_email` endpoint needs the client email address so we created this `useVerifyEmail` hook to populate the field with the email from `ClientStore`.

But now I see devs are adding more logic to these hooks which is not what we intended, Maybe we should rethink this idea or just use the `useFetch` and the `useRequest` hooks directly 🤔 Let's have a call on this later if you had time 🙇🏻

Reply:
I see, thanks for clarification man :pray: 
sure man, let's have a call when you're ready.